homeassistant/components/blueprint/websocket_api.py

```diff
@@ -84,3 +91,83 @@ async def ws_import_blueprint(hass, connection, msg):
             },
         },
     )
+
+
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "blueprint/save",
+        vol.Required("domain"): cv.string,
+        vol.Required("path"): cv.path,
+        vol.Required("data"): cv.string,
+        vol.Optional("source_url"): cv.url,
+    }
+)
+async def ws_save_blueprint(hass, connection, msg):
+    """Save a blueprint."""
+
+    path = msg["path"]
+    domain = msg["domain"]
+
+    def create_file():
+        """Create blueprint file."""
+        blueprint_path = pathlib.Path(
+            hass.config.path(BLUEPRINT_FOLDER, domain, f"{path}.yaml")
+        )
+        if blueprint_path.exists():
+            raise FileAlreadyExists()
+
+        blueprint = models.Blueprint(yaml.parse_yaml(msg["data"]))
+        if "source_url" in msg:
+            blueprint.update_metadata(source_url=msg["source_url"])
+
+        blueprint_path.parent.mkdir(parents=True, exist_ok=True)
+        blueprint_path.write_text(blueprint.yaml())
+
+    try:
+        await hass.async_add_executor_job(create_file)
+    except FileAlreadyExists:
+        connection.send_error(msg["id"], "already_exists", "File already exists")
+        return
+    except HomeAssistantError as err:
+        connection.send_error(msg["id"], websocket_api.ERR_INVALID_FORMAT, str(err))
+        return
+    except OSError as err:
+        connection.send_error(msg["id"], websocket_api.ERR_UNKNOWN_ERROR, str(err))
+        return
+
+    connection.send_result(
+        msg["id"],
+    )
+
+
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "blueprint/delete",
+        vol.Required("domain"): cv.string,
+        vol.Required("path"): cv.path,
+    }
+)
+async def ws_delete_blueprint(hass, connection, msg):
+    """Delete a blueprint."""
+
+    path = msg["path"]
+    domain = msg["domain"]
+
+    blueprint_path = pathlib.Path(hass.config.path(BLUEPRINT_FOLDER, domain, path))
+
+    try:
+        await hass.async_add_executor_job(blueprint_path.unlink)
+    except OSError as err:
+        connection.send_error(msg["id"], websocket_api.ERR_UNKNOWN_ERROR, str(err))
+        return
```

Comment:
I think that we should make this a helper function on `DomainBlueprints` that we call from this command.